commit 97e02ad23392f929862d6a04643e7f2fd91baebc
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Sun Aug 27 22:10:17 2017 +0200

    hurd: fix build with -fstack-protector-strong
    
    libmachuser and libhurduser also need stack_chk_fail_local and they do not
    link against libc_nonshared.
    
           * mach/stack_chk_fail_local.c: New file.
           * hurd/stack_chk_fail_local.c: New file.
           * mach/Machrules ($(interface-library)-routines): Add
           stack_chk_fail_local.

diff --git a/hurd/stack_chk_fail_local.c b/hurd/stack_chk_fail_local.c
new file mode 100644
index 0000000000..305871fbc0
--- /dev/null
+++ b/hurd/stack_chk_fail_local.c
@@ -0,0 +1 @@
+#include <debug/stack_chk_fail_local.c>
diff --git a/mach/Machrules b/mach/Machrules
index 36adfc7681..29114b1eef 100644
--- a/mach/Machrules
+++ b/mach/Machrules
@@ -221,7 +221,7 @@ endif
 
 ifdef interface-library
 
-$(interface-library)-routines = $(interface-routines)
+$(interface-library)-routines = $(interface-routines) stack_chk_fail_local
 extra-libs += $(interface-library)
 extra-libs-others += $(interface-library)
 
diff --git a/mach/stack_chk_fail_local.c b/mach/stack_chk_fail_local.c
new file mode 100644
index 0000000000..305871fbc0
--- /dev/null
+++ b/mach/stack_chk_fail_local.c
@@ -0,0 +1 @@
+#include <debug/stack_chk_fail_local.c>
